File: AMS_电子对账/code/client/source/YssElecReco/Service/Er/IErBbInfoService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using FAST.Common.Service.Services.Base;
using YssElecReco.Pojo.Er;
using FAST.Common.Service.Attributes;
using YssElecReco.Context;

namespace YssElecReco.Service.Er
{
    /// <summary>
    /// 浏览电子对账信息service
    /// chenyoulong 20121226
    /// v1.0.0.7
    /// </summary>
    [ServiceAttribute(ServiceId = ElecrecoServiceIdConstant.OSGI_ELECRECO, ServiceController = "com.yss.uco.elecreco.support.controller.IErBbInfoServiceController")]
    public interface IErBbInfoService : IServiceBus
    {
        /// <summary>
        /// 修改电子对账信息审核状态、发送状态
        /// </summary>
        /// <param name="bbInfoList">bbInfoList</param>
        /// <param name="status">status</param>
        /// <returns>string</returns>
        [MethodAttribute]
        string updateBbInfo(List<ErBbInfo> bbInfoList, string status);

        /// <summary>
        /// 修改电子对账信息审核状态、发送状态
        /// </summary>
        /// <param name="bbInfoList">bbInfoList</param>
        /// <returns>string</returns>
        [MethodAttribute]
        string deleteBbInfo(List<ErBbInfo> bbInfoList);

        /// <summary>
        /// 根据业务主键
        /// </summary>
        /// <param name="xErBbInfo">xErBbInfo</param>
        /// <returns>xml数据</returns>
        [MethodAttribute]
        string getXmlFile(ErBbInfo xErBbInfo);

        /// <summary>
        /// 发送数据
        /// </summary>
        ///<param name="bbInfoList">发送数据</param>
        /// <returns>结果</returns>
        [MethodAttribute]
        string sendBbInfo(List<ErBbInfo> bbInfoList);


        /// <summary>
        /// 发送数据
        /// </summary>
        ///<param name="bbInfoList">发送数据</param>
        /// <returns>string</returns>
        [MethodAttribute]
        string witeBbInfo(List<ErBbInfo> bbInfoList);

        /// <summary>
        /// 根据物理主键获得对象
        /// </summary>
        /// <param name="id">id</param>
        /// <returns>ErBbInfo</returns>
        [MethodAttribute]
        ErBbInfo getBbInfoById(string id);

        /// <summary>
        /// 重启对账管理
        /// </summary>
        /// <returns>1</returns>
        [MethodAttribute]
        string reStartDzMgr();

        /// <summary>
        /// STORY31217【招商财富】电子对账管理已反馈新增一个人工可以修改处理状态的功能
        /// 修改数据为对账一致
        /// </summary>
        ///<param name="bbInfoList">发送数据</param>
        /// <returns>string</returns>
        [MethodAttribute]
        string acceptBbInfo(List<ErBbInfo> bbInfoList);

        /// <summary>
        /// STORY54447反馈不一致结果人工对账一致处理需填写原因与说明信息 
        /// 新增其他对账中的对账一致方法
        /// </summary>
        ///<param name="bbInfoList">发送数据</param>
        /// <returns>string</returns>
        [MethodAttribute]
        string acceptBbInfoForQTDZ(List<ErBbInfo> bbInfoList);


        
        /// <summary>
        /// 操作不对账组合
        /// </summary>
        /// <param name="bbInfoList">bbInfoList</param>
        /// <param name="operType">operType</param>
        /// <returns>string</returns>
        [MethodAttribute]
        string UnPortOper(List<ErBbInfo> bbInfoList, string operType);
        /// <summary>
        /// STORY #65389电子对账前台与估值核算解耦
        /// </summary>
        /// <param name="pojos"></param>
        /// <param name="execOperCode"></param>
        /// <param name="isCheckExe"></param>
        /// <returns></returns>
        [MethodAttribute]
        string lockEconfirm(List<ErBbInfo> bbInfoList, string execOperCode, string isCheckExe, string executeType);


        /// <summary>
        /// wSTORY #77811 要可以撤销手工设置的对帐一致
        /// 修改或删除对账一致数据
        /// </summary>
        ///<param name="bbInfoList">发送数据</param>
        /// <returns>string</returns>
        [MethodAttribute]
        string unAcceptClick(List<ErBbInfo> bbInfoList);
        /// <summary>
        /// wSTORY #77811 要可以撤销手工设置的对帐一致
        /// 查询差异行数
        /// </summary>
        /// <param name="bbInfoList"></param>
        /// <returns></returns>
        [MethodAttribute]
        string queryNumberOfRows(ErBbInfo erBbInfo);

    }
}

File: AMS_电子对账/code/server/source/com.yss.uco.elecreco.support/src/main/java/com/yss/uco/elecreco/support/service/IErBbInfoService.java
package com.yss.uco.elecreco.support.service;

import java.util.Date;
import java.util.List;
import java.util.Map;

import com.yss.framework.api.mvc.biz.IServiceBus;
import com.yss.framework.api.restful.annotations.LinkControllerMethod;
import com.yss.framework.api.restful.annotations.LinkControllerMethodArgu;
import com.yss.framework.api.restful.annotations.RestfulSupported;
import com.yss.uco.elecreco.support.bean.ErBbInfo;
import com.yss.uco.elecreco.support.vo.ErBbInfoVo;

@RestfulSupported
public interface IErBbInfoService extends IServiceBus {

	/**
	 * 修改电子对账信息审核状态、发送状态
	 * @param bbInfoList
	 * @param status
	 * @return
	 */
	@LinkControllerMethod(value = "updateBbInfo", arguTypes = { ErBbInfoVo.class })
	public String updateBbInfo(@LinkControllerMethodArgu("bbInfoList")List<ErBbInfo> bbInfoList,@LinkControllerMethodArgu("status")String status);
	
	/**
	 * 删除电子对账信息
	 * @param bbInfoList
	 * @return
	 */
	public String deleteBbInfo(List<ErBbInfo> bbInfoList);
	
	/**
	 * 根据报文序号获得发送的xml文件
	 * @param fsn
	 * @return
	 */
	String getXmlFile(ErBbInfo erBbInfo);
	
	String sendBbInfo(List<ErBbInfo> bbInfoList);
	
//	String witeBbInfo(List<ErBbInfo> bbInfoList);
	
	ErBbInfo getBbInfoById(String id);
	////重启对账管理器
	String reStartDzMgr();
	
	/**
	 * STORY31217【招商财富】电子对账管理已反馈新增一个人工可以修改处理状态的功能
	 * @param bbInfoList 
	 * @return
	 */
	String acceptBbInfo(List<ErBbInfo> pojoList);
	
	/**
	 * STORY54447反馈不一致结果人工对账一致处理需填写原因与说明信息 新增其他对账中的对账一致方法
	 * @param bbInfoList 
	 * @return
	 */
	String acceptBbInfoForQTDZ(List<ErBbInfo> pojoList);
	
	/**
	 * STORY #50374 电子对账功能优化 
	 * 操作不对账组合
	 * @param pojoList operType：add-设置为不对账，remove-解除不对账设置 
	 */
	@LinkControllerMethod(value = "UnPortOper", arguTypes = { ErBbInfoVo.class })
	String UnPortOper(@LinkControllerMethodArgu("bbInfoList")List<ErBbInfo> pojoList,@LinkControllerMethodArgu("operType")String operType);
	/**
	 * 净值锁定
	 * @param pojos
	 * @param execOperCode
	 * @param isCheckExe
	 * @return
	 */
	@LinkControllerMethod(value = "lockEconfirm", arguTypes = { ErBbInfoVo.class })
	String lockEconfirm(@LinkControllerMethodArgu("bbInfoList")List<ErBbInfo> pojos,@LinkControllerMethodArgu("execOperCode")String execOperCode,@LinkControllerMethodArgu("isCheckExe")String isCheckExe,@LinkControllerMethodArgu("executeType")String executeType);
	
	/**
	 * 获取电子对账结果信息
	 * @param dDate
	 * @param assCodes 资产代码
	 * @return
	 */
	@LinkControllerMethod(value = "getDzResultInfo", arguTypes = { ErBbInfoVo.class })
	public Map<String, ErBbInfo> getDzResultInfo(@LinkControllerMethodArgu("dDate")Date dDate, @LinkControllerMethodArgu("assCodes")String assCodes);

	/**
	 * 是否是人工一致的数据
	 * @param csn 报文序号
	 * @return
	 */
	public boolean isManualAccept(String csn);
	
	/**
     * wSTORY #77811 要可以撤销手工设置的对帐一致
     * 修改或删除对账一致数据
     * @param pojoList
     * @return
     */
	String unAcceptClick(List<ErBbInfo> pojoList);
    /**
     * wSTORY #77811 要可以撤销手工设置的对帐一致
     * 查询差异行数
     * @param pojoList
     * @return
     */
	 String queryNumberOfRows(ErBbInfo erBbInfo);
	 
}
